### PR TITLE
BibIndex: do not use hstRECORD

### PR DESCRIPTION
**Pull request by fpoli**
##### BibIndex: improve virtual index queue tests
- Refactors the regression test `BibIndexVirtualIndexQueueTableTest`,
  in order to keep the queue table clean even if a test fails.

Signed-off-by: Federico Poli federico.poli@cern.ch

---
##### BibUpload: factor out step 6
- Move out the job end dates updating in a separate function
  to be able to call it in tests.

---
##### BibIndex: test concurrent bibupload and bibindex

---
##### BibIndex: do not use hstRECORD …
- Disables the filtering step that uses `hstRECORD` table:
  set always `force_all_indexes=True` when calling
  `find_affected_records_for_index`
- Fixes `find_affected_records_for_index` when called with `recIDs=[]`
- Fixes some regression test that relied on the utilization of
  `hstRECORD` in BibIndex

Signed-off-by: Federico Poli federico.poli@cern.ch

### CODE REVIEW

File: modules/bibindex/lib/bibindex_regression_tests.py [hunk 1]
```diff
@@ -1938,8 +1939,8 @@ def test_3_correct_entry_in_queue_for_phrase_table(self):
         self.index_dependent_index('keyword', [[19,19]], CFG_BIBINDEX_INDEX_TABLE_TYPE["Phrases"])
         query = "SELECT * FROM idxPHRASE01Q"
         res = run_sql(query)
-        self.assertEqual((19, 19), (res[0][2], res[0][3]))
-        self.assertEqual('keyword', res[0][4])
+        content = [(row[2], row[3], row[4]) for row in res]
+        self.assertIn((19, 19, 'keyword'), content)
```
Comment (kaplun) [inline modules/bibindex/lib/bibindex_regression_tests.py]: What are we correcting here?

File: modules/bibindex/lib/bibindex_engine.py [hunk 2]
```diff
@@ -2137,9 +2138,7 @@ def task_run_core():
     recIDs_range = get_recIDs_from_cli(regular_indexes)
     recIDs_for_index = find_affected_records_for_index(regular_indexes,
                                                        recIDs_range,
-                                                       (task_get_option("force") or \
-                                                       task_get_option("reindex") or \
-                                                       task_get_option("cmd") == "del"))
+                                                       True)
```
Comment (kaplun) [inline modules/bibindex/lib/bibindex_engine.py:2148]: We should comment this with a FIXME, since we are going to revert this change as soon as we introduce the new column with the dedicated timestamp.

File: modules/bibindex/lib/bibindex_regression_tests.py (hunk 1 above)
Comment (fpoli) [inline modules/bibindex/lib/bibindex_regression_tests.py]: I think that disabling the `hstRECORD` lookup bibindex indexes more records, so there are more tokens in the queue and the `(19, 19, 'keyword')` row is no more the first one.
Maybe is better to check that the number of occurences of `(19, 19, 'keyword')` increased?

**Comment by jalavik** [comment]
This is good to go? @Osso @fpoli  @kaplun

**Comment by kaplun** [comment]
I guess so

**Comment by kaplun** [comment]
I'll take care of it...

**Comment by fpoli** [comment]
The workaround is ready, yes. On Asana is https://app.asana.com/0/11661778102949/15477310167434
